amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonldschema/parser/JsonLDObjectElementParser.scala

```diff
@@ -39,11 +39,19 @@ case class JsonLDObjectElementParser(
   override def findClassTerm(ctx: SemanticContext): Seq[String] = {
     val terms = super.findClassTerm(ctx).map(ctx.expand)
     if (terms.isEmpty) {
-      val fragment = if (path.toString.isEmpty) key else path.toString
+      val fragment = lastNameFromPath() match {
```

Comment:
this whole if-clause could be refactored into a method so we isolate @id computation logic